setup.py

```diff
@@ -1,54 +1,71 @@
-#!/usr/bin/env python
-"""Setup file for python-matlab-bridge"""
-
+# -*- coding: utf-8 -*-
+"""
+Pymatbridge: A python interface to call out to Matlab(R)
+"""
 import os
-import sys
 import shutil
 import glob
 
-
 # BEFORE importing distutils, remove MANIFEST. distutils doesn't properly
 # update it when the contents of directories change.
-if os.path.exists('MANIFEST'):
-    os.remove('MANIFEST')
+if os.path.exists('MANIFEST'): os.remove('MANIFEST')
 
+# Set Version Info
+exec(open('pymatbridge/__version__.py').read())
 
-from distutils.core import setup
-
+try:
+    from setuptools import  setup
+except ImportError:
+    from distutils.core import setup
 
 # Find the messenger binary file(s) and copy it to /matlab folder.
 from messenger.make import get_messenger_dir
-messenger_dir = get_messenger_dir()
-
-for f in glob.glob("./messenger/%s/messenger.*" % messenger_dir):
-    shutil.copy(f, "./pymatbridge/matlab")
-
-
-# Get version and release info, which is all stored in pymatbridge/version.py
-ver_file = os.path.join('pymatbridge', 'version.py')
-exec(open(ver_file).read())
-
-opts = dict(name=NAME,
-            maintainer=MAINTAINER,
-            maintainer_email=MAINTAINER_EMAIL,
-            description=DESCRIPTION,
-            long_description=LONG_DESCRIPTION,
-            url=URL,
-            download_url=DOWNLOAD_URL,
-            license=LICENSE,
-            classifiers=CLASSIFIERS,
-            author=AUTHOR,
-            author_email=AUTHOR_EMAIL,
-            platforms=PLATFORMS,
-            version=VERSION,
-            packages=PACKAGES,
-            package_data=PACKAGE_DATA,
-            requires=REQUIRES,
-            #extras_require=EXTRAS_REQUIRE,
-            scripts=BIN
-            )
 
+for f in glob.glob("messenger/%s/messenger.*" % get_messenger_dir()):
+    shutil.copy(f, 'pymatbridge/matlab/')
 
 # Now call the actual setup function
 if __name__ == '__main__':
-    setup(**opts)
+    setup(
```

Comment:
Yeah - it's rather well thought-through. In particular, it plays very well
with sphinx, to build nice documentation pages, which is something we
should do one day...

On Sun, Mar 29, 2015 at 9:21 AM, Sang Han <notifications@github.com> wrote:

> In setup.py
> <https://github.com/arokem/python-matlab-bridge/pull/172#discussion_r27355909>
> :
>
> >
> >  # Now call the actual setup function
> >  if __name__ == '__main__':
> > -    setup(**opts)
> > +    setup(
>
> Sorry, I forgot about these changes, they should be on another brach!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arokem/python-matlab-bridge/pull/172/files#r27355909>.
>